### ISSUE TITLE
[SCRAM] Generate hashes of the password on the frontend and send it over to the backend within the register-user flow

### ISSUE THREAD

**Issue by interestIngc** [globe-and-citizen/layer8#152]
Implement password hashes generation on the frontend and send it over to the backend, see the diagram below:
<img width="849" alt="Screenshot 2024-12-08 at 19 10 54" src="https://github.com/user-attachments/assets/c53a0203-db52-4ba4-bf33-987258b510d3">

More specifically, we should:
1. Assume that `iterationCount` and `salt` were returned in the server response from `register-user-precheck` endpoint (check [Issue #151](https://github.com/globe-and-citizen/layer8/issues/151) for more details), and mock it for now to unblock development of the final step in register-user flow.
2. In `server/assets-v1/templates/src/pages/user_portal/register.html ` generate `storedKey` and `serverKey`:
 ```
saltedPassword = pbkdf2(password, salt, iterationCount)
clientKey = HMAC-SHA256(saltedPassword, `Client Key`)
serverKey = HMAC-SHA256(saltedPassword, `Server Key`)
storedKey = SHA256(clientKey)
```
Look up about [HMAC](https://en.wikipedia.org/wiki/HMAC) and [PBKDF2](https://en.wikipedia.org/wiki/PBKDF2) here.
3. Send `storedKey` and `serverKey`, as well as other user specific data (`username, first_name, last_name, display_name, country, public_key`) to the backend as the request to `/api/v1/register-user` endpoint.
4. Modify the `/api/v1/register-user` endpoint:
   * Save user specific data, `storedKey` and `serverKey` into the `users` table instead of the password. We should make it via the update query, because row representing the user was already added in [Issue #151](https://github.com/globe-and-citizen/layer8/issues/151). 
   * Remove other logic such as salt generation since it's already done in the precheck endpoint.
5. Add tests to cover the newly implemented logic, i.e. changes in `controller.go`, `service.go` and `repository.go`.
6. Add migration scripts to modify the schema of `users` table, i.e. add new columns `storedKey` and `serverKey`.
